### PR TITLE
Model `java.util.Properties.setProperty`

### PR DESCRIPTION
**Pull request by intrigus-lgtm**
Models `java.util.Properties.setProperty` as mentioned in https://github.com/github/codeql/pull/9817#issuecomment-1183619419

### CODE REVIEW

**Comment by intrigus-lgtm** [comment]
I don't know whether this deserves a change note and whether this could be merged with the existing change note ([2022-07-13-properites.md](https://github.com/github/codeql/pull/9817/files#diff-a17ad305a88c8668d2027c7bdc6d9040272de64507161b1979d890f9eb404cfa)) as the change note seems to have already been released (https://github.com/github/codeql/blob/main/java/ql/lib/change-notes/released/0.3.1.md#minor-analysis-improvements).
Note that the existing change note both has a typo in the filename "properites" and in the content "java.util.Properites".

File: java/ql/test/library-tests/dataflow/collections/Test.java [hunk 1]
```diff
@@ -84,7 +84,17 @@ public void run4() {
...
     sink(p.getProperty("key", "defaultValue")); // Flow
-    
+
+    Properties clean = new Properties();
+    sink(clean.getProperty("key", tainted)); // Flow
+  }
+
+  public void run5() {
+    Properties p = new Properties();
+    p.setProperty("key", tainted);
```
Comment (smowton) [inline java/ql/test/library-tests/dataflow/collections/Test.java:94]: You should test that the return value of setProperty returns taint from the map too

**Review by smowton — COMMENTED** [review]
Can't merge change notes, so yes give this a new one

File: java/ql/test/library-tests/dataflow/collections/Test.java (hunk 1 above)
Comment (intrigus-lgtm) [inline java/ql/test/library-tests/dataflow/collections/Test.java:94]: Like this?
```suggestion
    sink(p.setProperty("key", tainted)); // Flow
```

File: java/ql/test/library-tests/dataflow/collections/Test.java [hunk 2]
```diff
@@ -84,7 +84,17 @@ public void run4() {
     p.put("key", tainted);
```
Comment (intrigus-lgtm) [inline java/ql/test/library-tests/dataflow/collections/Test.java:84]: This method is similar to `setProperty`, should I also change it like this?
```suggestion
    sink(p.put("key", tainted)); // Flow
```

File: java/ql/test/library-tests/dataflow/collections/Test.java (hunk 1 above)
Comment (smowton) [reply java/ql/test/library-tests/dataflow/collections/Test.java:94]: I believe it returns the *old* value, so you actually want to store some taint, then store something clean and check that the return value of the clean store shows taint

File: java/ql/test/library-tests/dataflow/collections/Test.java (hunk 2 above)
Comment (intrigus-lgtm) [reply java/ql/test/library-tests/dataflow/collections/Test.java:84]: Done in https://github.com/github/codeql/pull/9963/commits/8eebe710ce568c49f57c51d958b211e4e0e4fa63

File: java/ql/test/library-tests/dataflow/collections/Test.java (hunk 1 above)
Comment (intrigus-lgtm) [reply java/ql/test/library-tests/dataflow/collections/Test.java:94]: Yes, you're right, `setProperty` returns the *old* value.
Done in https://github.com/github/codeql/pull/9963/commits/8eebe710ce568c49f57c51d958b211e4e0e4fa63

Comment (intrigus-lgtm) [reply java/ql/test/library-tests/dataflow/collections/Test.java:94]: Done in https://github.com/github/codeql/pull/9963/commits/8eebe710ce568c49f57c51d958b211e4e0e4fa63

File: java/ql/test/library-tests/dataflow/collections/Test.java [hunk 3]
```diff
@@ -84,8 +84,30 @@ public void run4() {
...
     sink(p.getProperty("key")); // Flow
     sink(p.getProperty("key", "defaultValue")); // Flow
-    
+
+    Properties clean = new Properties();
+    sink(clean.getProperty("key", tainted)); // Flow
```
Comment (smowton) [inline java/ql/test/library-tests/dataflow/collections/Test.java:89]: This is accidentally duplicated on lines 98-99

Comment (intrigus-lgtm) [reply java/ql/test/library-tests/dataflow/collections/Test.java]: Deduplicated :)

**Review by smowton — APPROVED** [review]

**Pull request merged**